src/main/java/io/debezium/connector/yugabytedb/YugabyteDBConnectorConfig.java

```diff
@@ -1433,6 +1534,165 @@ private static int validateLogicalDecodingMessageExcludeList(Configuration confi
+                    String query;
+                    if (!checkIfPublicationExists(publicationName, ybConnection)) {
+                        query = "CREATE PUBLICATION " + publicationName + " FOR ALL TABLES;";
+                        try (Connection connection = ybConnection.connection()) {
+                            ybConnection.execute(query);
+                        }
+                    } else {
+                        LOGGER.info("The publication {} already exists", publicationName);
+                    }
+                    return fetchTableIncludeList(ybConnection, publicationName);
+                } else if (autoCreateMode.equalsIgnoreCase("filtered")) {
+                    String tableList = config.getString(TABLE_INCLUDE_LIST);
+                    if (tableList == null || tableList.isEmpty()) {
+                        throw new DebeziumException("No tables included for Filtered Publication");
+                    }
+
+                    String createQuery = "CREATE PUBLICATION " + config.getString(PUBLICATION_NAME) + " FOR TABLE " + tableList + ";";
+                    String alterQuery = "ALTER PUBLICATION " + config.getString(PUBLICATION_NAME) + " SET TABLE " + tableList + ";";
+                    Boolean createNeeded = !checkIfPublicationExists(publicationName, ybConnection);
+                    
+                    try (Connection connection = ybConnection.connection();
+                          final Statement statement = connection.createStatement()) {
+                        statement.execute(createNeeded ? createQuery : alterQuery);
+                    } 
+                    return tableList;
+                } else {
+                    throw new DebeziumException("Invalid Autocreate Mode");
+                }
+            } catch (SQLException e) {
+                retryCount++;
+                exception = e;
+                if (retryCount > config.getInteger(MAX_CONNECTOR_RETRIES)) {
+                    LOGGER.error("Failed to extract table list from publication. All {} retries failed",config.getInteger(MAX_CONNECTOR_RETRIES), e);
+                    throw new DebeziumException(e);
+                }
+                LOGGER.warn("Error while trying to extract table list from publication. Will retry, attempt {} out of {}",
+                            retryCount, config.getInteger(MAX_CONNECTOR_RETRIES));
+                e.printStackTrace();
+
+                try {
+                    final Metronome retryMetronome = Metronome.parker(Duration.ofMillis(config.getLong(CONNECTOR_RETRY_DELAY_MS)), Clock.SYSTEM);
+                    retryMetronome.pause();
+                } catch (InterruptedException ie) {
+                    LOGGER.warn("Connector retry sleep interrupted by exception: {}", ie);
+                    Thread.currentThread().interrupt();
+                }
+            } catch (DebeziumException e) {
+                exception = e;
```

Comment:
This catch statement is redundant here. Please remove.

Reply:
We are throwing `DebeziumException` in certain cases like when autocreate is disabled and publication does not exist. We don't want to retry in these cases. That is why I had added a separate catch block for this.

Reply:
Yeah, but you are not doing anything here.

```
catch (some_exception) {
  // an assignment
  throw some_exception
}
```

The above part doesn't have any effect.

Reply:
Catching and rethrowing an exception = Not catching it